istioctl/cmd/istioctl/verifyinstall.go

```diff
@@ -0,0 +1,330 @@
+	resources := make([]*unstructured.Unstructured, 0, len(chunks))
+	for i, chunk := range chunks {
+		chunk = bytes.TrimSpace(chunk)
+		if len(chunk) == 0 {
+			continue
+		}
+		u, err := parseChunk(chunk)
+		if err != nil {
+			log.Debugf("Error processing [%d]: %v", i, err)
+			continue
+		}
+		if u == nil {
+			continue
+		}
+		resources = append(resources, u)
+	}
+	return resources, nil
+}
+
+func parseChunk(chunk []byte) (*unstructured.Unstructured, error) {
+	u := &unstructured.Unstructured{}
+	if err := yaml.Unmarshal(chunk, u); err != nil {
+		return nil, err
+	}
+	if empty(u) {
+		return nil, nil
+	}
+	return u, nil
+}
+
+// Check if the parsed resource is empty
+func empty(r *unstructured.Unstructured) bool {
+	if r.Object == nil || len(r.Object) == 0 {
+		return true
+	}
+	return false
+}
+
+// GetDeploymentCondition returns the condition with the provided type.
+func GetDeploymentCondition(status v1beta1.DeploymentStatus, condType v1beta1.DeploymentConditionType) *v1beta1.DeploymentCondition {
+	for i := range status.Conditions {
+		c := status.Conditions[i]
+		if c.Type == condType {
+			return &c
+		}
+	}
+	return nil
+}
+
+func getDeploymentStatus(deployment *v1beta1.Deployment, name string) error {
+	cond := GetDeploymentCondition(deployment.Status, v1beta1.DeploymentProgressing)
+	if cond != nil && cond.Reason == "ProgressDeadlineExceeded" {
+		return fmt.Errorf("deployment %q exceeded its progress deadline", name)
+	}
+	if deployment.Spec.Replicas != nil && deployment.Status.UpdatedReplicas < *deployment.Spec.Replicas {
+		return fmt.Errorf("waiting for deployment %q rollout to finish: %d out of %d new replicas have been updated",
+			name, deployment.Status.UpdatedReplicas, *deployment.Spec.Replicas)
+	}
+	if deployment.Status.Replicas > deployment.Status.UpdatedReplicas {
+		return fmt.Errorf("waiting for deployment %q rollout to finish: %d old replicas are pending termination",
+			name, deployment.Status.Replicas-deployment.Status.UpdatedReplicas)
+	}
+	if deployment.Status.AvailableReplicas < deployment.Status.UpdatedReplicas {
+		return fmt.Errorf("waiting for deployment %q rollout to finish: %d of %d updated replicas are available",
+			name, deployment.Status.AvailableReplicas, deployment.Status.UpdatedReplicas)
+	}
+	return nil
+}
+func buildKubeClient(kubeconfig string) error {
+	restConfig, err := kube.BuildClientConfig(kubeconfig, configContext)
+
+	if err != nil {
+		return err
+	}
+	kubeClient, err = kclient.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	extensionClient, err = apiextensionsclient.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateInputFlags() error {
+	var err error
+	if installFileName == "" {
+		err = multierr.Append(err, errors.New("--filename must be set"))
+	}
+	return err
+}
+func reportErrorOrLogOutStatus(err error, resourceType string, resourceFullName string) error {
+	if err != nil {
```

Comment:
The way you're calling this, the error can be due to a transient error in the client, as well as an actual error. You may want to distinguish between them.